plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiModule.java

```diff
@@ -76,7 +80,27 @@ public void configure(Binder binder)
     @Provides
     public ExecutorService createExecutorService()
     {
-        return newCachedThreadPool(daemonThreadsNamed("hudi-split-manager-%d"));
+        return newCachedThreadPool(daemonThreadsNamed("hudi-split-manager-%s"));
+    }
+
+    @ForHudiSplitSource
+    @Singleton
+    @Provides
+    public ScheduledExecutorService createSplitLoaderExecutor(HudiConfig hudiConfig)
```

Comment:
Why do we need a `ScheduledExecutorService`? It seems to only be used as a regular `ExecutorService`.

Also, why do we need two separate thread pools, both configured with the same concurrency?

Reply:
Added two separate thread pools due to the nature of work being done. One task is simply adding partitions to the async queue, the other task is listing those partitions which is IO bound. I kept the default concurrency same for both as it should not matter much for small table (few tens or hundreds of partitions) but for larger tables they should be tuned. It's a way to balance the tradeoff. When working with larger tables with thousands of partitions, it's better to increase the `splitGeneratorParallelism`. 
You're right about `ScheduledExecutorService`. I think we started with that but don't see a need.